gsa/src/gmp/commands/users.js

```diff
@@ -267,6 +270,38 @@ class UserCommand extends EntityCommand {
     });
   }
 
+  getReportComposerDefaults() {
+    return this.httpGet({
+      cmd: 'get_setting',
+      setting_id: REPORT_COMPOSER_DEFAULTS_SETTING_ID,
+    }).then(response => {
+      const {data} = response;
+      const {setting = {}} = data.get_settings.get_settings_response;
+      const {value} = setting;
+      let defaults;
+
+      try {
+        defaults = JSON.parse(value);
+      }
+      catch (e) {
+        log.warn('Could not parse report composer defaults');
+        return;
```

Comment:
Maybe return sane defaults for the reportcomposerdefaults here?

Reply:
That would silently swallow the error disregarding that something is wrong in the database. In case the settings can't be loaded, sane defaults are set in the props for DownloadReportDialog and TriggerAlertDialog. If any of the defaults is undefined, they'll get set there.

Reply:
I think your code will crash currently in this case because you are expecting a response object in your action creator (access to response.data) and will get only undefined. Instead of returning sane defaults via return response.setData(...) you could also throw an exception and therefore cause the promise catch path to create an error action.